src/docs/guides/csp-nonce.md

```diff
@@ -0,0 +1,117 @@
+
+### Output Targets
+
+Using nonces may differ slightly between output targets, so please be sure to use the correct pattern based on the context in which your
+Stencil components are consumed.
+
+#### Dist
+
+Consuming a `nonce` in the `dist` output target is easy using the provided `setNonce` helper function. This function is exported from the index
+file of the output target's designated output directory.
+
+This function simply accepts the `nonce` string value that you want set for every `style` and `script` tag.
+
+This is an example of consuming the `dist` output in an Angular app's entrypoint:
+
+```ts
+// main.ts
+
+import { defineCustomElements, setNonce } from 'my-lib/loader';
+
+// Will set the `nonce` attribute for all scripts/style tags
+// i.e will run styleTag.setAttribute('nonce', 'r4nd0m')
+// Obviously, you should use the nonce generated by your server
+setNonce('r4nd0m');
+
+// Generic Angular bootstrapping
+platformBrowserDynamic()
+  .bootstrapModule(AppModule)
+  .catch(err => console.log(err));
+
+defineCustomElements();
+```
+
+#### Custom Elements
+
+Consuming a `nonce` in the `dist-custom-elements` output target is easy using the provided `setNonce` helper function. This function is exported
+from the index file of the output target's designated output directory.
+
+This function simply accepts the `nonce` string value that you want set for every `style` and `script` tag.
+
+This is an example of consuming the `dist-custom-elements` output in an Angular app's entrypoint:
+
+```ts
+// main.ts
+
+import { defineCustomElements, setNonce } from 'my-lib/dist/components';
+// We'll assume we have `autoDefineCustomElements` enabled
+import 'my-lib/dist/components/my-component';
+
+// Will set the `nonce` attribute for all scripts/style tags
+// i.e will run styleTag.setAttribute('nonce', 'r4nd0m')
+// Obviously, you should use the nonce generated by your server
+setNonce('r4nd0m');
+
+// Generic Angular bootstrapping
+platformBrowserDynamic()
+  .bootstrapModule(AppModule)
+  .catch(err => console.log(err));
+```
+
+#### WWW
+
+Unfortunately, setting `nonce` attributes gets a bit trickier when it comes to [SSR and SSG](docs/static-site-generation). As a `nonce` needs
+to be unique per page view, we cannot define/set the nonce at build time. So, this responsibility now falls on the
+[hydrate app](docs/hydrate-app)'s execution of runtime code.
+
+##### SSR
+
+Since there is not an easy way (or any way) of exposing and executing helper functions to manipulate the outcome of the runtime code, Stencil
+has fallback behavior for pulling the `nonce` off of the window object.
+
+So, for SSR, your app can simply set the desired `nonce` value on `window.nonce` at bootstrap. Yes, this does involve some manual configuration
+for the code served by your server.
```

Comment:
Can we update this to clarify that 'bootstrap' should be when the page is rendered? I'd like to make it clear that this shouldn't just happen once, but every time a server serves the page.